katsdpimageutils/primarybeamcorrection/primary_beam_correction.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import os
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from astropy import wcs
+from astropy import coordinates
+
+
+def read_fits(pathfits):
+    """
+    This functions reads in the fits file
+    Input: Path to the fits file
+    Output: Fits file object
+    """
+    fl = fits.open(pathfits)
+    images = fl[0]
+    return images
+
+
+def get_positions(path):
+    """
+    This function reads the fits file and returns the sky coordinate
+    of the phase centre.
+    Input: path to the fits file
+    Output: Sky coordinate of the phase centre
+    """
+    image_wcs = wcs.WCS(path)
+    # Get pointing centre of observation
+    phase_centre_ra = image_wcs.celestial.wcs.crval[0]
```

Comment:
Also, are you wanting the pointing centre or the phase centre (the comment says one, the variable name says the other). I forget what the use case is, but I think the commissioners have in the past run observations where they're not the same (maybe @spassmoor can shed some light on it?).

For a general FITS file the crval could be different to both of them (e.g. if it was reprojected); you'd have to ask @mauch what plane the continuum pipeline projects its facets to.

Reply:
`crval` of the spatial axes of katsdpcontim images will always be whatever the astrometric radec of the `katpoint.Target` is from the input `katdal` object - presumably the phase centre of the observation.
There is no information stored in the FITS headers about the commanded antenna pointing position. We could probably add this as a separate (nonstandard) FITS header from available sensors in such a way that it won't change if someone re-projects the pixels.
@bmerry what does katsdpimager's beam correction do if pointing centre != phase centre?

Reply:
With regard to science imaging , I do not know of any LSP or planned science cases that will have  pointing centre != phase centre.  There are no commissioning plans to test such a mode either.

Reply:
@SihlanguI I think just update the docstring to say pointing centre (because that's what you ideally want it to return), but add a note that the implementation assumes this to be the same as the crval. Actually fixing things to handle the general case is a problem for another day (or year).